pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/TransactionBuffer.java

```diff
@@ -160,6 +160,12 @@ public interface TransactionBuffer {
      */
     PositionImpl getMaxReadPosition();
 
+    /**
+     * Get the can dispatch max position.
+     * @return the stable position.
+     */
+    CompletableFuture<Position> getLastCanDispatchPosition();
```

Comment:
Maybe we don't need this method, and move `TopicTransactionBuffer#getLastCanDispatchPosition`'s logic to `PersistentTopic#getLastCanDispatchPositionWithTxn()` is better.